File: SEMESTRE 1/PROG1/MATERIAL SOLE/EJERCICIOS COMPARTIDOS/Archivos compartidos/factorial.cpp
#include <program1.h>
/**
*    Función recursiva que calcula el factorial de un entero no negativo.
*/

funcion largo fact(entero n) {
largo resul;
si(n > 0) entonces                                              // Paso recursivo
    resul = n * fact(n-1);
    sinoSi(n ES 0) entonces                                     // Caso base 
    resul = 1;
    sino                                                        // Error
    finSi
retorna(resul);
}

File: SEMESTRE 1/LAB1/FizzBuzz/src/fizzbuzz/Factorial.java
package fizzbuzz;

public class Factorial {

    public static int Factorial(int n){
        int result = 1;
               
        for (int i = 1; i < n+1; i++) {
            result  *= i;
        }
        
        return result;
    }
    
    public static int FactorialRecursive(int n){
        int fact = 0;
        
        if (n==0) {
            fact = 1;
        }else{
            fact=n*FactorialRecursive(n-1);
        }
        
        return fact;
    }
    
}
